packages/quantic/force-app/main/default/lwc/quanticRefineToggle/quanticRefineToggle.js

```diff
@@ -0,0 +1,207 @@
+ *   <div slot="refine-title">Custom Title</div>
+ * </c-quantic-refine-toggle>
+ */
+export default class QuanticRefineToggle extends LightningElement {
+  labels = {
+    sortAndFilters,
+    viewResults,
+  };
+
+  /**
+   * The ID of the engine instance the component registers to.
+   * @api
+   * @type {string}
+   */
+  @api engineId;
+  /**
+   * Indicates whether the icon of the refine toggle button should be hidden.
+   * @api
+   * @type {string}
+   */
+  @api hideIcon;
+  /**
+   * The label to be shown in the refine toggle button.
+   * @api
+   * @type {string}
+   * @defaultValue `'Sort & Filters'`
+   */
+  @api buttonLabel = this.labels.sortAndFilters;
+  /**
+   * Indicates whether the Quantic Sort component should be hidden.
+   * @api
+   * @type {boolean}
+   */
+  @api hideSort;
+  /**
+   * Indicates whether the refine modal will be opened in full screen.
+   * @api
+   * @type {boolean}
+   * @defaultValue `false`
+   */
+  @api fullScreen = false;
+
+  /** @type {QuerySummary} */
+  querySummary;
+  /** @type {BreadcrumbManager} */
+  breadcrumbManager;
+  /** @type {number} */
+  total;
+  /** @type {number} */
+  activeFiltersCount = 0;
+  /** @type {string} */
+  modalId = 'refineModal';
+
+  connectedCallback() {
+    registerComponentForInit(this, this.engineId);
+  }
+
+  renderedCallback() {
+    initializeWithHeadless(this, this.engineId, this.initialize);
+  }
+
+  /**
+   * @param {SearchEngine} engine
+   */
+  initialize = (engine) => {
+    this.querySummary = CoveoHeadless.buildQuerySummary(engine);
+    this.breadcrumbManager = CoveoHeadless.buildBreadcrumbManager(engine);
+
+    this.unsubscribeQuerySummary = this.querySummary.subscribe(() =>
+      this.updateTotalResults()
+    );
+    this.unsubscribeBreadcrumbManager = this.breadcrumbManager.subscribe(() =>
+      this.updateActiveFiltersCount()
+    );
+  };
+
+  disconnectedCallback() {
+    this.unsubscribeQuerySummary?.();
+    this.unsubscribeBreadcrumbManager?.();
+  }
+
+  /**
+   * Updates total number of results.
+   * @returns {void}
+   */
+  updateTotalResults() {
+    this.total = this.querySummary.state.total;
+  }
+
+  /**
+   * Updates the number of active filters.
+   * @returns {void}
+   */
+  updateActiveFiltersCount() {
+    this.activeFiltersCount = this.breadcrumbManager.state.hasBreadcrumbs
+      ? this.getFiltersCountFromAllBreadcrumbs()
+      : 0;
+  }
+
+  /**
+   * Returns the number of active filters from all the breadcrumbs.
+   * @returns {number}
+   */
+  getFiltersCountFromAllBreadcrumbs() {
+    const facetBreadcrumbKeys = [
+      'facetBreadcrumbs',
+      'numericFacetBreadcrumbs',
+      'categoryFacetBreadcrumbs',
+      'dateFacetBreadcrumbs',
+      'staticFilterBreadcrumbs',
+    ];
+    return facetBreadcrumbKeys.reduce(
+      (count, facetKey) =>
+        count +
+        this.getFiltersCountFromFacetBreadcrumb(
+          this.breadcrumbManager.state[facetKey]
```

Comment:
If, for some reason, some keys are not defined in the state, the running total would become `NaN`. You should include a fallback to add `0` in such case.

Reply:
sure 👌